src/storyMap/components/StoryMap.js

```diff
@@ -0,0 +1,543 @@
+
+import _ from 'lodash/fp';
+import { useTranslation } from 'react-i18next';
+import scrollama from 'scrollama';
+
+import { Box, Link } from '@mui/material';
+
+import RichTextEditor from 'common/components/RichTextEditor';
+
+import mapboxgl from 'gis/mapbox';
+import { chapterHasVisualMedia } from 'storyMap/storyMapUtils';
+
+import { MAPBOX_ACCESS_TOKEN } from 'config';
+
+import { ALIGNMENTS, LAYER_TYPES } from '../storyMapConstants';
+
+import './StoryMap.css';
+
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+
+const CURRENT_LOCATION_CHECK_PRESSISION = 13; // 13 decimal places
+
+const transformRequest = url => {
+  const hasQuery = url.indexOf('?') !== -1;
+  const suffix = hasQuery
+    ? '&pluginName=scrollytellingV2'
+    : '?pluginName=scrollytellingV2';
+  return {
+    url: url + suffix,
+  };
+};
+
+const getBoundsJson = bounds => ({
+  type: 'FeatureCollection',
+  features: [
+    {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [bounds._sw.lng, bounds._sw.lat],
+            [bounds._ne.lng, bounds._sw.lat],
+            [bounds._ne.lng, bounds._ne.lat],
+            [bounds._sw.lng, bounds._ne.lat],
+            [bounds._sw.lng, bounds._sw.lat],
+          ],
+        ],
+      },
+    },
+  ],
+});
+
+const Audio = ({ record }) => {
+  return (
+    <>
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+      <audio style={{ width: '100%' }} controls>
+        <source src={record.media.signedUrl} type={record.media.type} />
+      </audio>
+    </>
+  );
+};
+
+const Image = ({ record }) => {
+  return <img src={record.media.signedUrl} alt={record.title}></img>;
+};
+
+const Chapter = ({ theme, record }) => {
+  const { t } = useTranslation();
+  const classList = [
+    'step-container',
+    'step',
+    ALIGNMENTS[record.alignment] || 'centered',
+    ...(record.hidden ? ['hidden'] : []),
+  ].join(' ');
+
+  const hasVisualMedia = chapterHasVisualMedia(record);
+
+  return (
+    <Box
+      id={record.id}
+      component="section"
+      aria-label={t('storyMap.view_chapter_label', { title: record.title })}
+      className={classList}
+    >
+      <Box
+        className={`${theme} step-content`}
+        sx={{
+          width: hasVisualMedia ? '50vw' : 'auto',
+        }}
+      >
+        {record.title && (
+          <h3 id={`title-${record.id}`} className="title">
+            {record.title}
+          </h3>
+        )}
+        {record.media &&
+          (record.media.type.startsWith('image') ? (
+            <Image record={record} />
+          ) : record.media.type.startsWith('audio') ? (
+            <Audio record={record} />
+          ) : record.media.type.startsWith('embedded') ? (
+            <iframe
+              title={record.media.title}
+              src={record.media.url}
+              frameborder="0"
+              style={{ height: '300px', width: '100%' }}
```

Comment:
@paulschreiber This is just an arbitrary value to avoid it rendering bigger and using more space, what is your suggestion for this value?

Reply:
Is this one desktop or mobile? If mobile, do we want 100vw (or similar)?

Reply:
@paulschreiber This is defining the iframe width it will use 100% of the width of the step container, this is not changing the behavior of how it looks in mobile